src/java.desktop/windows/native/libawt/windows/ShellFolder2.cpp

```diff
@@ -719,12 +720,14 @@ JNIEXPORT jlong JNICALL Java_sun_awt_shell_Win32ShellFolder2_getLinkLocation
     switch (strret.uType) {
       case STRRET_CSTR :
         // IShellFolder::ParseDisplayName requires the path name in Unicode.
-        MultiByteToWideChar(CP_ACP, MB_PRECOMPOSED, strret.cStr, -1, olePath, MAX_PATH);
+        ret = MultiByteToWideChar(CP_ACP, MB_PRECOMPOSED, strret.cStr, -1, olePath, MAX_PATH);
+        if (ret == 0) return 0;
```

Comment:
if (ret == 0) {
    return NULL;
}

Since (1) We always use parens, (2) the other cases return NULL .. unless you want to also change all those to return 0.